test_scripts/API/VehicleData/GearStatus/common.lua

```diff
@@ -0,0 +1,441 @@
+    m.getHMIConnection():SendResponse(data.id, data.method, "SUCCESS", { gearStatus = params })
+  end)
+  m.getMobileSession():ExpectResponse(cid, { success = false, resultCode = "GENERIC_ERROR" })
+end
+
+--[[ @sendOnVehicleData: Processing OnVehicleData RPC
+--! @parameters:
+--! pData: parameters for the notification
+--! pExpTime: expected number of notifications
+--! pParam: parameters for OnVehicleData RPC
+--! pAppID: application number (1, 2, etc.)
+--! @return: none
+--]]
+function m.sendOnVehicleData(pData, pExpTime, pParam, pAppID)
+  if not pExpTime then pExpTime = 1 end
+  if not pParam then pParam = "gearStatus" end
+  if not pAppID then pAppID = 1 end
+  m.getHMIConnection():SendNotification("VehicleInfo.OnVehicleData", { [pParam] = pData })
+  m.getMobileSession(pAppID):ExpectNotification("OnVehicleData", { [pParam] = pData })
+  :Times(pExpTime)
+end
+
+--[[ @ignitionOff: IGNITION_OFF sequence
+--! @parameters: none
+--! @return: none
+--]]
+function m.ignitionOff()
+  config.ExitOnCrash = false
+  local timeout = 5000
+  local function removeSessions()
+    for i = 1, m.getAppsCount() do
+      m.deleteSession(i)
+    end
+  end
+  local event = events.Event()
+  event.matches = function(event1, event2) return event1 == event2 end
+  EXPECT_EVENT(event, "SDL shutdown")
+  :Do(function()
+    removeSessions()
+    StopSDL()
+    config.ExitOnCrash = true
+  end)
+  m.getHMIConnection():SendNotification("BasicCommunication.OnExitAllApplications", { reason = "SUSPEND" })
+  m.getHMIConnection():ExpectNotification("BasicCommunication.OnSDLPersistenceComplete")
+  :Do(function()
+    m.getHMIConnection():SendNotification("BasicCommunication.OnExitAllApplications",{ reason = "IGNITION_OFF" })
+    for i = 1, m.getAppsCount() do
+      m.getMobileSession(i):ExpectNotification("OnAppInterfaceUnregistered", { reason = "IGNITION_OFF" })
+    end
+  end)
+  m.getHMIConnection():ExpectNotification("BasicCommunication.OnAppUnregistered", { unexpectedDisconnect = false })
+  :Times(m.getAppsCount())
+  local isSDLShutDownSuccessfully = false
+  m.getHMIConnection():ExpectNotification("BasicCommunication.OnSDLClose")
+  :Do(function()
+    utils.cprint(35, "SDL was shutdown successfully")
+    isSDLShutDownSuccessfully = true
+    RAISE_EVENT(event, event)
+  end)
+  :Timeout(timeout)
+  local function forceStopSDL()
+    if isSDLShutDownSuccessfully == false then
+      utils.cprint(35, "SDL was shutdown forcibly")
+      RAISE_EVENT(event, event)
+    end
+  end
+  RUN_AFTER(forceStopSDL, timeout + 500)
+end
+
+--[[ @unexpectedDisconnect: closing connection
+--! @parameters: none
+--! @return: none
+--]]
+function m.unexpectedDisconnect()
+  m.getHMIConnection():ExpectNotification("BasicCommunication.OnAppUnregistered", { unexpectedDisconnect = true })
+  :Times(actions.mobile.getAppsCount())
+  actions.mobile.disconnect()
+  utils.wait(1000)
+end
+
+--[[ @checkResumption_FULL: function that checks HMI level resumption to FULL level
```

Comment:
in review for other VD was proposed to check only data resumption and remove checks for HMI level resumption